xclim/testing/tests/test_indices.py

```diff
@@ -413,6 +413,112 @@ def test_effective_growing_degree_days(
+                "MS",
+                1,
+                "gamma",
+                "APP",
+                [1.3166409, 1.4506897, 1.9460845, -3.09, 0.8506797],
+                2.6e-05,
+            ),
+            (
+                "MS",
+                12,
+                "gamma",
+                "APP",
+                [0.5982093, 1.5597635, 1.6930935, 0.9963997, 0.7027995],
+                7.4e-05,
+            ),
+            ("MS", 1, "gamma", "ML", [1.317, 1.451, 1.946, -3.09, 0.851], 0.19),
+            ("MS", 12, "gamma", "ML", [0.5982, 1.5598, 1.6931, 0.9964, 0.7028], 0.034),
+        ],
+    )
+    def test_standardized_precipitation_index(
+        self, freq, window, dist, method, values, diff_tol
+    ):
+
+        # Only location=1 is tested
+        ds = open_dataset("sdba/CanESM2_1950-2100.nc").isel(location=1)
+        pr = ds.pr.sel(time=slice("2000"))
+        pr_cal = ds.pr.sel(time=slice("1950", "1980"))
+        spi = xci.standardized_precipitation_index(
+            pr, pr_cal, freq, window, dist, method
+        )
+
+        # Only a few moments before year 2000 are tested
+        spi = spi.isel(time=slice(-11, -1, 2))
+
+        # [Guttman, 1999]: The number of precipitation events (over a month/season or
+        # other time period) is generally less than 100 in the US. This suggests that
+        # bounds of ± 3.09 correspond to 0.999 and 0.001 probabilities. SPI indices outside
+        # [-3.09, 3.09] might be non-statistically relevant. In `climate_indices` the SPI
+        # index is clipped outside this region of value. In the values chosen above,
+        # this doesn't play role, but let's clip it anyways to avoid future problems.
+        # The last few values in time are tested
+        spi = spi.clip(-3.09, 3.09)
+
+        np.testing.assert_allclose(spi.values, values, rtol=0, atol=diff_tol)
+
+    # See SPI version
+    @pytest.mark.parametrize(
+        "freq, window, dist, method,  values, diff_tol",
+        [
+            ("MS", 1, "gamma", "APP", [1.505, 1.753, 2.334, -2.484, 0.879], 0.2),
+            ("MS", 12, "gamma", "APP", [0.6012, 1.6908, 1.8678, 0.9801, 0.6566], 0.024),
+            ("MS", 1, "gamma", "ML", [1.505, 1.753, 2.334, -2.484, 0.879], 0.76),
+            ("MS", 12, "gamma", "ML", [0.6012, 1.6908, 1.8678, 0.9801, 0.6566], 0.038),
+        ],
+    )
+    def test_standardized_precipitation_evapotranspiration_index(
+        self, freq, window, dist, method, values, diff_tol
+    ):
+        print(values)
+        # Only location=1 is tested
+        ds = (
+            open_dataset("sdba/CanESM2_1950-2100.nc")
+            .isel(location=1)
+            .sel(time=slice("2000"))
+        )
+        pr = ds.pr
+        # generate water budget
+        with xr.set_options(keep_attrs=True):
+            tasmax = ds.tasmax
+            tas = tasmax - [2 + 1 * 0.5 for i in range(ds.tasmax.shape[0])]
+            tasmin = tasmax - [4 + 2 * 0.5 for i in range(ds.tasmax.shape[0])]
```

Comment:
```suggestion
            tas = tasmax - 3.5
            tasmin = tasmax - 7
```
Or did I miss something?

Reply:
Earlier writing had an index dependent random number, but not anymore. Thanks